secure_redis/settings.py

```diff
@@ -4,18 +4,34 @@
 from django.core.exceptions import ImproperlyConfigured
 
 
-def get_secure_cache_opts():
-    if hasattr(settings, 'DJANGO_REDIS_SECURE_CACHE_NAME'):
-        cache_name = settings.DJANGO_REDIS_SECURE_CACHE_NAME
-    else:
+def get_secure_cache_name():
+    """
+    Return the cache name to use for django-rq functionality
+    """
+    if not hasattr(settings, 'DJANGO_REDIS_SECURE_CACHE_NAME'):
         cache_name = 'default'
+    else:
+        cache_name = settings.DJANGO_REDIS_SECURE_CACHE_NAME
+    return cache_name
 
-    secure_cache_options_settings = settings.CACHES[cache_name].get('OPTIONS')
-    if not secure_cache_options_settings:
+
+def get_secure_cache_opts():
```

Comment:
@ptink have you tested the library with your changes applied? I haven't tested it, but seems that we will have `AttributeError` in `serializer.py` in case  `DJANGO_REDIS_SECURE_CACHE_NAME ` is set to None. In this case, `get_secure_cache_opts()` will return None.

Reply:
I don't believe so? I've tested it and the behaviour is the same as it was previously

Reply:
So let us assume that `DJANGO_REDIS_SECURE_CACHE_NAME` is set to `None` in settings. Then the following will happen:

1. `serializers.py` will call `get_secure_cache_opts()` method to initialize `SecureSerializer`
2. `get_secure_cache_opts` method will call `get_secure_cache_name`
3. `get_secure_cache_name` will return None when settings.DJANGO_REDIS_SECURE_CACHE_NAME is set to None.
4. `get_secure_cache_opts` will return None because `cache_name` is None.
5. `SecureSerializer.__init__` will receive `options` parameter as None.
6. In `SecureSerializer.__init__` upon executing the line `param_key = options.get("REDIS_SECRET_KEY")`, an AttributeError will be raised because `options` in None, and we are trying to access a method `get` from a NoneType.

Reply:
Aha, I follow now and yes you're right. I missed that when testing - I think I must have had a settings override giving me a false positive. I'll push a fix, as an aside what's the exact command you use to run the tests? would be good to document it and I'll make sure everything runs on py3 as well